app/src/main/java/org/p2p/wallet/home/ui/main/HomeContract.kt

```diff
@@ -28,5 +28,6 @@ interface HomeContract {
         fun toggleTokenVisibility(token: Token.Active)
         fun toggleTokenVisibilityState()
         fun clearTokensCache()
+        fun hasUsername(): Boolean
```

Comment:
im not sure that presenter can have a username :) 

i can only suggest Davran's idea - put it all in the presenter
Presenter will have only one method - onProfileIconClicked or smth like this

and presenter will have the logic and the if inside. 
```
if (username exists)
view.navigateToProfile
else
view.navigateToReserveName
```
making the view stupid as hell

Reply:
Nice point 🤘